sml/src/sml_shared.c

```diff
@@ -17,9 +17,29 @@
 // along with libSML.  If not, see <http://www.gnu.org/licenses/>.
 
 #include <sml/sml_shared.h>
+#include <stdarg.h>
 #include <stdio.h>
 #include <string.h>
 
+void (*sml_error)(const char *format, ...);
+
+void sml_error_default(const char *format, ...) {
+	va_list args;
+	va_start(args, format);
+
+	char *format2 = malloc(9 + strlen(format) + 1 + 1);
```

Comment:
btw: why 9? the terminating zero is already at the end, or? So I'd expect
8 (= strlen("libsml: ") + ... + 1 (\n) + 1 (term. zero)